sky/provision/kubernetes/instance.py

```diff
@@ -676,14 +676,17 @@ def _create_namespaced_pod_with_retries(namespace: str, pod_spec: dict,
 def _create_persistent_volume_claim(namespace: str, context: Optional[str],
                                     pvc_spec: Dict[str, Any]) -> None:
     """Creates a persistent volume claim for SkyServe controller."""
-    try:
-        kubernetes.core_api(context).read_namespaced_persistent_volume_claim(
-            name=pvc_spec['metadata']['name'], namespace=namespace)
+    name = pvc_spec['metadata']['name']
+    field_selector = f'metadata.name={name}'
+    pvcs = (
+        kubernetes.core_api(context).list_namespaced_persistent_volume_claim(
+            namespace, field_selector=field_selector).items)
+    if pvcs:
+        logger.info(f'Updating PVC: {name}')
+        kubernetes.core_api(context).patch_namespaced_persistent_volume_claim(
+            name, namespace, pvc_spec)
```

Comment:
If a PVC with the name already exists and it was not created by SkyPilot, should we throw an error instead of silently patching it?